### PR TITLE
RELATED: SD-757 - Pushpin Chart has only location and without measure.

### PR DESCRIPTION
**Pull request by dlcd2203**
…efault config.

---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [x] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [x] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [x] Successful `extended test - examples`
- [x] Successful `extended test - storybook`

# Related PRs

- gdc-analytical-designer:
- gdc-dashboards:

# Related Jira tasks

SD-757: https://jira.intgdc.com/browse/FET-757

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5593/ : SUCCESS in 6m 39s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1125/ : SUCCESS in 12s

File: src/helpers/dimensions.ts [hunk 1]
```diff
@@ -304,15 +304,14 @@ function getScatterDimensions(mdObject: VisualizationObject.IVisualizationObject
...
-        },
-    ];
+    const { attributes = [], measures }: AFM.IAfm = convertBucketsToAFM(buckets);
+    const chartDimensions: AFM.IDimension[] = [];
+
+    if (measures) {
+        chartDimensions.push({ itemIdentifiers: [MEASUREGROUP] });
+    }
+    chartDimensions.push({ itemIdentifiers: attributes.map((attr: AFM.IAttribute) => attr.localIdentifier) });
+    return chartDimensions;
```
Comment (naibinh) [inline src/helpers/dimensions.ts]: Please add unit test for this function

File: src/helpers/dimensions.ts [hunk 2]
```diff
@@ -304,15 +304,14 @@ function getScatterDimensions(mdObject: VisualizationObject.IVisualizationObject
...
-            itemIdentifiers: [MEASUREGROUP],
-        },
-        {
-            itemIdentifiers: attributes.map((attr: AFM.IAttribute) => attr.localIdentifier),
-        },
-    ];
+    const { attributes = [], measures }: AFM.IAfm = convertBucketsToAFM(buckets);
+    const chartDimensions: AFM.IDimension[] = [];
+
+    if (measures) {
```
Comment (ptquang86) [inline src/helpers/dimensions.ts]: is `measures` an array or `undefined` in this case?
should we check for `measures.length` instead?

File: src/helpers/dimensions.ts [hunk 3]
```diff
@@ -304,15 +304,14 @@ function getScatterDimensions(mdObject: VisualizationObject.IVisualizationObject
...
+        chartDimensions.push({ itemIdentifiers: [MEASUREGROUP] });
+    }
+    chartDimensions.push({ itemIdentifiers: attributes.map((attr: AFM.IAttribute) => attr.localIdentifier) });
```
Comment (ptquang86) [inline src/helpers/dimensions.ts]: should we check for empty `attributes`?

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5598/ : SUCCESS in 11m 18s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1133/ : SUCCESS in 7s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5599/ : SUCCESS in 10m 39s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1134/ : SUCCESS in 5s

**Comment by dlcd2203** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1318.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

File: src/helpers/dimensions.ts (hunk 3 above)
Comment (dlcd2203) [reply src/helpers/dimensions.ts]: attributes define in internal part, so it will not be empty.

**Comment by gdgate** [comment]
Build in pipeline check-extended aborted.

File: src/helpers/tests/dimensions.spec.ts [hunk 4]
```diff
@@ -981,3 +985,56 @@ describe("getGeneralDimensionsFromAFM", () => {
         expect(getGeneralDimensionsFromAFM(afm)).toEqual(expectedDimensions);
     });
 });
+
+describe("getGeoChartDimensions", () => {
+    function createBuckets(pushpinProps: IGeoPushpinChartBucketProps): VisualizationObject.IBucket[] {
```
Comment (ptquang86) [inline src/helpers/tests/dimensions.spec.ts]: should we re-use the `getBuckets` function from `src/components/GeoPushpinChart.tsx`?

File: src/helpers/tests/dimensions.spec.ts [hunk 5]
```diff
@@ -981,3 +985,56 @@ describe("getGeneralDimensionsFromAFM", () => {
...
+        };
+        const buckets: VisualizationObject.IBucket[] = createBuckets(pushpinProps);
+        const expectedDimensions: AFM.IDimension[] = [
+            {
+                itemIdentifiers: ["location"],
+            },
+        ];
+        expect(getGeoChartDimensions(buckets)).toEqual(expectedDimensions);
+    });
+    it("should return for AFM with measureGroup Array and Attributes array have 2 item", () => {
```
Comment (ptquang86) [inline src/helpers/tests/dimensions.spec.ts]: `... items`

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5600/ : SUCCESS in 4m 42s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1136/ : SUCCESS in 5s

File: src/helpers/tests/dimensions.spec.ts (hunk 4 above)
Comment (dlcd2203) [reply src/helpers/tests/dimensions.spec.ts]: i just needs mock-data, so don't need to import getBuckets from geoPushpinchart component.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5602/ : SUCCESS in 6m 14s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1138/ : SUCCESS in 5s

**Comment by naibinh** [comment]
LGTM

**Comment by ptquang86** [comment]
LGTM

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5604/ : SUCCESS in 4m 54s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1140/ : SUCCESS in 4s

**Comment by dlcd2203** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1318.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5607/ : SUCCESS in 17m 35s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1143/ : SUCCESS in 7s

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1144/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2306/ : FAILURE in 21m 59s

**Comment by dlcd2203** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1318.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1146/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2307/ : FAILURE in 12m 40s

**Comment by dlcd2203** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1318.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1148/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2308/ : FAILURE in 15m 07s

**Comment by dlcd2203** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2879/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1149/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2309/ : SUCCESS in 16m 06s

**Comment by dlcd2203** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1318.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1150/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2310/ : SUCCESS in 10m 47s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2880/ : SUCCESS in 16m 11s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5610/ : SUCCESS in 5m 33s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1350/ : SUCCESS in 17s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/1868/ : SUCCESS in 12m 11s

**Pull request merged**